api/endpoints/upload.py

```
@@ -2,56 +2,116 @@
         parsing_status = "success"
-        text_preview = None
-        if file.content_type == "application/pdf":
+        text_preview: Optional[str] = None
+
+        if content_type == "application/pdf":
             try:
                 text = extract_text_from_pdf(saved_path)
                 text_preview = text[:500] if text else None
             except DocumentParseError:
-                logger.error("Document parse error")
+                logger.error("Document parse error", extra={"path": saved_path})
```

Ruff (0.13.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)